compiler/resolution/functionResolution.cpp

```diff
@@ -8928,10 +8928,9 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           call->insertAtTail(unrefCall);
           FnSymbol* unrefFn = resolveNormalCall(unrefCall);
           resolveFns(unrefFn);
-
-          if (unrefFn->retType == tmp->typeInfo()) {
-            // If the types are equal, we must be dealing with a non-view
-            // array, so we can remove the useless unref call.
+          if (unrefFn->where == NULL) {
```

Comment:
We could teach the compiler to identify an array view _array, but that seems like more work.
If we need to distinguish the array-view unref from the array one, I think a flag would be a reasonable way to do so, for now at least.

But, I'm not yet understanding why this change was necessary. Is there some pattern where we can create an array over a sliced domain e.g. where you wanted to change this behavior? Checking for the `where` clause doesn't seem right. Can you explain the problematic case that caused you to make this change?